hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/SerDeHelper.java

```diff
@@ -263,6 +273,31 @@ private static Type parseTypeFromJson(JsonNode jsonNode) {
     throw new IllegalArgumentException(String.format("cannot parse type from jsonNode: %s", jsonNode));
   }
 
+  private static Object parseDefaultValueFromJson(JsonNode jsonNode, Type type) {
+    if (!jsonNode.has(DEFAULT_VALUE)) {
+      return null;
+    }
+    String defaultValue = jsonNode.get(DEFAULT_VALUE).asText();
+    switch (type.typeId()) {
+      case BOOLEAN:
+        return Boolean.parseBoolean(defaultValue);
+      case INT:
+        return Integer.parseInt(defaultValue);
+      case LONG:
+        return Long.parseLong(defaultValue);
+      case FLOAT:
+        return Float.parseFloat(defaultValue);
+      case DOUBLE:
+        return Double.parseDouble(defaultValue);
+      case BINARY:
+        return Base64.getDecoder().decode(defaultValue);
+      case DECIMAL:
+        return new BigDecimal(defaultValue);
+      default:
+        return defaultValue;
```

Comment:
where is other type？
  private static Object parseDefaultValue(Type type, JsonNode field) {
    if (!field.has(DEFAULT_VALUE)) {
      return null;
    }
    Object defaultValue = null;
    switch (type.typeId()) {
      case RECORD:
      case ARRAY:
      case MAP:
        JsonNode defaultNode = field.get(DEFAULT_VALUE);
        defaultValue = JacksonUtils.toObject(defaultNode);
        break;
      case STRING:
        defaultValue = field.get(DEFAULT_VALUE).asText();
        break;
      case INT:
        defaultValue = field.get(DEFAULT_VALUE).asInt();
        break;
      case LONG:
        defaultValue = field.get(DEFAULT_VALUE).asLong();
        break;
      case FLOAT:
        defaultValue = field.get(DEFAULT_VALUE).asDouble();
        break;
      case DOUBLE:
        defaultValue = field.get(DEFAULT_VALUE).asDouble();
        break;
      case BOOLEAN:
        defaultValue = field.get(DEFAULT_VALUE).asBoolean();
        break;
      case DECIMAL:
        defaultValue = toBytes(field.get(DEFAULT_VALUE));
        break;
      case FIXED:
        defaultValue = toBytes(field.get(DEFAULT_VALUE));
        break;
      case BINARY:
        defaultValue = toBytes(field.get(DEFAULT_VALUE));
        break;
      case DATE:
        defaultValue = field.get(DEFAULT_VALUE).asInt();
        break;
      case TIME:
        defaultValue = field.get(DEFAULT_VALUE).asLong();
        break;
      case TIMESTAMP:
        defaultValue = field.get(DEFAULT_VALUE).asLong();
        break;
      case UUID:
        defaultValue = field.get(DEFAULT_VALUE).asText();
        break;
      default:
        throw new RuntimeException("not implemented");
    }
    return defaultValue;
  }